SVGPathSerializer.mm

```diff
@@ -131,6 +143,76 @@
     }
 }
 
+CGPathRef svgParser::readRectTag()
+{
+    NSCAssert(strcasecmp((char*)xmlTextReaderConstName(_xmlReader), "rect") == 0,
+              @"Not on a <polygon>");
+    
+    char *xDef = (char *)xmlTextReaderGetAttribute(_xmlReader, (xmlChar*)"x");
+    char *yDef = (char *)xmlTextReaderGetAttribute(_xmlReader, (xmlChar*)"y");
+    char *widthDef = (char *)xmlTextReaderGetAttribute(_xmlReader, (xmlChar*)"width");
+    char *heightDef = (char *)xmlTextReaderGetAttribute(_xmlReader, (xmlChar*)"height");
+    
+    CGRect rectRect = CGRectMake(atof(xDef), atof(yDef), atof(widthDef), atof(heightDef));
+    
+    free(xDef);
+    free(yDef);
+    free(widthDef);
+    free(heightDef);
+    
+    NSMutableString *pathAttribute = [[NSMutableString alloc] init];
+    [pathAttribute appendFormat:@"M %f %f ", CGRectGetMinX(rectRect), CGRectGetMinY(rectRect)];
+    [pathAttribute appendFormat:@"H %f ", CGRectGetMaxX(rectRect)];
+    [pathAttribute appendFormat:@"V %f ", CGRectGetMaxY(rectRect)];
+    [pathAttribute appendFormat:@"H %f ", CGRectGetMinX(rectRect)];
+    [pathAttribute appendFormat:@"V %f Z", CGRectGetMinY(rectRect)];
+    
+    CGPathRef const path = pathDefinitionParser(pathAttribute).parse();
+    
+    if(!path) {
+        NSLog(@"*** Error: Invalid path attribute");
+        return NULL;
+    } else {
+        return path;
+    }
+    
+}
+
+CGPathRef svgParser::readPolygonTag()
+{
+    NSCAssert(strcasecmp((char*)xmlTextReaderConstName(_xmlReader), "polygon") == 0,
+              @"Not on a <polygon>");
+    
+    char *pathDef = (char *)xmlTextReaderGetAttribute(_xmlReader, (xmlChar*)"points");
+    
+    NSString *pointsString = [[NSString alloc] initWithCString:pathDef encoding:NSUTF8StringEncoding];
+    NSMutableArray *items = [[pointsString componentsSeparatedByCharactersInSet:[NSCharacterSet characterSetWithCharactersInString:@", "]] mutableCopy];
+    
+    if([items count] < 1) {
```

Comment:
Didn't notice this till now, but right below this check, *2* items are read from the array.